syft/pkg/cataloger/deb/cataloger.go

```diff
@@ -56,93 +59,166 @@ func (c *Cataloger) Catalog(resolver source.FileResolver) ([]pkg.Package, error)
-			if err != nil {
-				return nil, fmt.Errorf("unable to find dpkg md5 contents: %w", err)
-			}
+			// fetch additional data from the copyright file to derive the license information
+			addLicenses(resolver, dbLocation, p)
+		}
 
-			if md5Reader != nil {
-				// attach the file list
-				metadata.Files = parseDpkgMD5Info(md5Reader)
-
-				// keep a record of the file where this was discovered
-				if md5Location != nil {
-					p.Locations = append(p.Locations, *md5Location)
-				}
-			} else {
-				// ensure the file list is an empty collection (not nil)
-				metadata.Files = make([]pkg.DpkgFileRecord, 0)
-			}
+		results = append(results, pkgs...)
+	}
+	return results, nil
+}
 
-			// persist alterations
-			p.Metadata = metadata
+func addLicenses(resolver source.FileResolver, dbLocation source.Location, p *pkg.Package) {
+	// get license information from the copyright file
+	copyrightReader, copyrightLocation := fetchCopyrightContents(resolver, dbLocation, p)
 
-			// get license information from the copyright file
-			copyrightReader, copyrightLocation, err := fetchCopyrightContents(resolver, dbLocation, p)
-			if err != nil {
-				return nil, fmt.Errorf("unable to find dpkg copyright contents: %w", err)
-			}
+	if copyrightReader != nil {
+		// attach the licenses
+		p.Licenses = parseLicensesFromCopyright(copyrightReader)
+
+		// keep a record of the file where this was discovered
+		if copyrightLocation != nil {
+			p.Locations = append(p.Locations, *copyrightLocation)
+		}
+	}
+}
 
-			if copyrightReader != nil {
-				// attach the licenses
-				p.Licenses = parseLicensesFromCopyright(copyrightReader)
+func mergeFileListing(resolver source.FileResolver, dbLocation source.Location, p *pkg.Package) {
+	metadata := p.Metadata.(pkg.DpkgMetadata)
 
-				// keep a record of the file where this was discovered
-				if copyrightLocation != nil {
-					p.Locations = append(p.Locations, *copyrightLocation)
-				}
+	// get file listing (package files + additional config files)
+	files, infoLocations := getAdditionalFileListing(resolver, dbLocation, p)
+loopNewFiles:
+	for _, newFile := range files {
+		for _, existingFile := range metadata.Files {
+			if existingFile.Path == newFile.Path {
+				// skip adding this file since it already exists
+				continue loopNewFiles
 			}
 		}
+		metadata.Files = append(metadata.Files, newFile)
+	}
 
-		results = append(results, pkgs...)
+	// sort files by path
+	sort.SliceStable(metadata.Files, func(i, j int) bool {
+		return metadata.Files[i].Path < metadata.Files[j].Path
+	})
+
+	// persist alterations
+	p.Metadata = metadata
+
+	// persist location information from each new source of information
+	p.Locations = append(p.Locations, infoLocations...)
+}
+
+func getAdditionalFileListing(resolver source.FileResolver, dbLocation source.Location, p *pkg.Package) ([]pkg.DpkgFileRecord, []source.Location) {
+	// ensure the file list is an empty collection (not nil)
+	var files = make([]pkg.DpkgFileRecord, 0)
```

Comment:
I think we've gone back-and-forth on this before, but what is the benefit of creating a slice of `0` instead of (I know the linter would complain):

```suggestion
	var files []pkg.DpkgFileRecord
```

Reply:
I can update the comment to be more specific on the reasoning here...

The reason for the allocation is for later presentation. That is, the metadata objects to not have redundant definitions, and the common team consensus on displaying collections in the JSON presenter is to always show empty collections (e.g. `{}`) instead of `nil`. This requires that objects with collections that will be presented have allocations in advance. The comment was meant to portray this, but only does so partially --I'll adjust.